packages/product-components/src/components/TokenSummaryIcon/TokenSummaryIcon.tsx

```diff
@@ -0,0 +1,99 @@
+import { AssetLogo, Row } from '@trezor/components';
+import { getCoingeckoId } from '@suite-common/wallet-config';
+import { EnhancedTokenInfo, getContractAddressForNetwork } from '@suite-common/token-definitions';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+
+import styled from 'styled-components';
+
+export type TokenSummaryIconProps = {
+    network: NetworkSymbol;
+    tokens: EnhancedTokenInfo[];
+};
+
+const Icon2or3Container = styled.div`
+    display: grid;
+    grid-template-columns: repeat(auto-fit, minmax(5px, max-content));
+    direction: rtl;
+    justify-items: end;
+`;
+
+const Icon4Container = styled.div`
+    display: grid;
+    grid-template-columns: repeat(auto-fit, minmax(4px, max-content));
+    direction: rtl;
+    justify-items: end;
+`;
+
+export const TokenSummaryIcon = ({ network, tokens }: TokenSummaryIconProps) => {
+    const { length } = tokens;
+    if (length > 3) {
+        tokens = tokens.slice(0, 3);
+    }
+
+    if (length) {
+        const coingeckoId = getCoingeckoId(network);
+
+        switch (length) {
+            case 1:
+                return (
+                    <AssetLogo
+                        size={24}
+                        coingeckoId={coingeckoId ?? ''}
+                        contractAddress={getContractAddressForNetwork(network, tokens[0].contract)}
+                        placeholder="ETH"
+                    />
+                );
+            case 2:
+            case 3:
+                return (
+                    <Row>
+                        <Icon2or3Container>
+                            {tokens
+                                ?.slice()
+                                .reverse()
+                                .map(token => (
+                                    <AssetLogo
+                                        key={token.contract}
+                                        size={16}
+                                        coingeckoId={coingeckoId ?? ''}
+                                        contractAddress={getContractAddressForNetwork(
+                                            network,
+                                            token.contract,
+                                        )}
+                                        placeholder="ETH"
+                                    />
+                                ))}
+                        </Icon2or3Container>
+                    </Row>
+                );
+
+            default:
```

Comment:
default and case 2 and 3 can be together. Container can accept number of tokens, empty token can be added just in case the length is gte 4.
Slicing and reversing can be then done outside the switch